src/interaction/MovablePoint.tsx

```diff
@@ -30,109 +27,11 @@ export function MovablePoint({
   constrain = (point) => point,
   color = Theme.pink,
 }: MovablePointProps) {
-  const { viewTransform, userTransform } = useTransformContext()
-  const { xSpan, ySpan } = useSpanContext()
-  const inverseViewTransform = vec.matrixInvert(viewTransform)
-  invariant(inverseViewTransform, "The view transform must be invertible.")
-
-  const inverseTransform = React.useMemo(() => getInverseTransform(userTransform), [userTransform])
-
-  const combinedTransform = React.useMemo(
-    () => vec.matrixMult(viewTransform, userTransform),
-    [viewTransform, userTransform],
-  )
-
-  const [dragging, setDragging] = React.useState(false)
-  const [displayX, displayY] = vec.transform(point, combinedTransform)
-
-  const pickup = React.useRef<vec.Vector2>([0, 0])
-
   const ref = React.useRef<SVGGElement>(null)
 
-  useDrag(
-    (state) => {
-      const { type, event } = state
-      event?.stopPropagation()
-
-      const isKeyboard = type.includes("key")
-      if (isKeyboard) {
-        event?.preventDefault()
-        const { direction: yDownDirection, altKey, metaKey, shiftKey } = state
-
-        const direction = [yDownDirection[0], -yDownDirection[1]] as vec.Vector2
-        const span = Math.abs(direction[0]) ? xSpan : ySpan
-
-        let divisions = 50
-        if (altKey || metaKey) divisions = 200
-        if (shiftKey) divisions = 10
-
-        const min = span / (divisions * 2)
-        const tests = range(span / divisions, span / 2, span / divisions)
+  const { dragging } = useMovementInteraction({ gestureTarget: ref, onMove, point, constrain })
```

Comment:
Feels really nice. I think passing a `ref` is also generally the norm for things like this so that makes me happy.